### PR TITLE
Pricing page rework: Add product/plan lightbox descriptions.

### PR DESCRIPTION
**Pull request by elliottprogrammer**
#### Proposed Changes

This PR adds the specific product descriptions that are displayed in the product/plan lightbox.

It also fixes an issue where the product lightbox descriptions were not displaying for some products, such as for Backup T2, Security T2, Anti-spam, Scan, Search, Boost, and Social.

Completes asana task: 1202796695664022-as-1202979995370873

- I also renamed the functions & properties for getting the product featured descriptions.  They were named/called productFeatured**Text**, which is inconsistent with the product _description_, product short _description_, product lightbox _description_, etc.  So I renamed "productFeatured**Text**" to "productFeatured**Description**", which is more descriptive and consistent with all the other product descriptions.

#### Testing Instructions

**View the regression:**

- Go to https://cloud.jetpack.com/pricing?flags=jetpack/pricing-page-rework-v1
- Click on "More about Backup" link and then select storage option "1 TB".  Notice that the lightbox product description (just under the product title) disappears.
- Click on the "More about <product>" lightbox links for Anti-spam, Scan, Search, Boost, and Social, and notice that the lightbox product descriptions are missing (not displaying).

**Test this PR:**

- Do any one of these
    * Click on Jetpack Cloud live link below and add this to the end of URL `?flags=jetpack/pricing-page-rework-v1`
    * or spin up this PR 
        * Run `git fetch && git checkout add/jetpack-products-lightbox-description`
        * Run `yarn start-jetpack-cloud`
        * Goto `http://jetpack.cloud.localhost:3000/pricing/?flags=jetpack/pricing-page-rework-v1`
     * Click on the "More about <product>" links for all the products/bundles (except not CRM Entrepreneur)
     * Verify all the lightbox product descriptions are displaying, (be sure to select the storage options variants for Backup & Security)
     * You can compare the lightbox product descriptions copy with this **Figma File**: m0y1jHTuDpvMPkFsOJBoNP-fi-3557%3A123824

#### Screenshots:

Markup 2022-09-14 at 16 08 55

Markup 2022-09-14 at 16 09 44

Markup 2022-09-14 at 16 10 39

Markup 2022-09-14 at 16 11 33

Markup 2022-09-14 at 16 12 16

Markup 2022-09-14 at 16 13 13

Markup 2022-09-14 at 16 13 49

Markup 2022-09-14 at 16 14 38

Markup 2022-09-14 at 16 15 37

Markup 2022-09-14 at 16 16 20

Markup 2022-09-14 at 16 16 54

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [X] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202979995370865
  - https://app.asana.com/0/0/1202979995370873

### CODE REVIEW

**Review by andrii-lysenko — APPROVED** [review]
Looks great and work as described. Even though I see that `getFeaturedProductText` and `getFeaturedPlanText` I didn't see it affecting anything or providing bugs.

LGTM!

File: packages/calypso-products/src/get-jetpack-product-featured-description.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ * Get Jetpack product featured text based on the product purchase object.
+ */
+export function getJetpackProductFeaturedDescription(
+	product: Product
+): TranslateResult | undefined {
+	const jetpackProductsFeaturedDescription = getJetpackProductsFeaturedDescription() as Record<
+		string,
+		TranslateResult
+	>;
```
Comment (manzoorwanijk) [inline packages/calypso-products/src/get-jetpack-product-featured-description.tsx]: Casting is not needed. Let us let TypeScript do what it does best - [inference](https://www.typescriptlang.org/docs/handbook/type-inference.html).

```suggestion

/**
 * Get Jetpack product featured text based on the product purchase object.
 */
export function getJetpackProductFeaturedDescription( product: Product ) {
	const jetpackProductsFeaturedDescription = getJetpackProductsFeaturedDescription();
```

File: packages/calypso-products/src/get-jetpack-product-lightbox-description.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ * Get Jetpack product lightbox description based on the product purchase object.
+ */
+export function getJetpackProductLightboxDescription(
+	product: Product
+): TranslateResult | undefined {
+	const jetpackProductsLightboxDescription = getJetpackProductsLightboxDescription() as Record<
+		string,
+		TranslateResult
+	>;
```
Comment (manzoorwanijk) [inline packages/calypso-products/src/get-jetpack-product-lightbox-description.ts]:
```suggestion

/**
 * Get Jetpack product lightbox description based on the product purchase object.
 */
export function getJetpackProductLightboxDescription( product: Product ) {
	const jetpackProductsLightboxDescription = getJetpackProductsLightboxDescription();
```

**Review by manzoorwanijk — APPROVED** [review]
Looks great, just a couple of minor code improvements.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7528982

Thank you @elliottprogrammer for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7528982).

**Pull request merged**